libs/oracledb/langchain_oracledb/vectorstores/oraclevs.py

```diff
@@ -0,0 +1,1929 @@
+        table_name: The name of the table to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the table.
+    """
+    connection = _get_connection(client)
+    if connection is None:
+        raise ValueError("Failed to acquire a connection.")
+    if _table_exists(connection, table_name):
+        with connection.cursor() as cursor:
+            ddl = f"DROP TABLE {table_name} PURGE"
+            cursor.execute(ddl)
+        logger.info("Table dropped successfully...")
+    else:
+        logger.info("Table not found...")
+    return
+
+
+@_handle_exceptions
+async def adrop_table_purge(client: Any, table_name: str) -> None:
+    """Drop a table and purge it from the database.
+
+    Args:
+        client: The OracleDB connection object.
+        table_name: The name of the table to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the table.
+    """
+
+    async def context(connection: Any) -> None:
+        if await _atable_exists(connection, table_name):
+            with connection.cursor() as cursor:
+                ddl = f"DROP TABLE {table_name} PURGE"
+                await cursor.execute(ddl)
+            logger.info("Table dropped successfully...")
+        else:
+            logger.info("Table not found...")
+
+    await _handle_context(client, context)
+    return
+
+
+@_handle_exceptions
+def drop_index_if_exists(client: Any, index_name: str) -> None:
+    """Drop an index if it exists.
+
+    Args:
+        client: The OracleDB connection object.
+        index_name: The name of the index to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the index.
+    """
+    connection = _get_connection(client)
+    if connection is None:
+        raise ValueError("Failed to acquire a connection.")
+    if _index_exists(connection, index_name):
+        drop_query = f"DROP INDEX {index_name}"
+        with connection.cursor() as cursor:
+            cursor.execute(drop_query)
+            logger.info(f"Index {index_name} has been dropped.")
+    else:
+        logger.exception(f"Index {index_name} does not exist.")
+    return
+
+
+@_handle_exceptions
+async def adrop_index_if_exists(client: Any, index_name: str) -> None:
+    """Drop an index if it exists.
+
+    Args:
+        client: The OracleDB connection object.
+        index_name: The name of the index to drop.
+
+    Raises:
+        RuntimeError: If an error occurs while dropping the index.
+    """
+
+    async def context(connection: Any) -> None:
+        if await _aindex_exists(connection, index_name):
+            drop_query = f"DROP INDEX {index_name}"
+            with connection.cursor() as cursor:
+                await cursor.execute(drop_query)
+                logger.info(f"Index {index_name} has been dropped.")
+        else:
+            logger.exception(f"Index {index_name} does not exist.")
+
+    await _handle_context(client, context)
+    return
+
+
+def get_processed_ids(
```

Comment:
Are these ids supposed to be unique as they are placed in primary column.

Reply:
Yes, we expect them to be unique.